README.rst

```diff
@@ -317,6 +317,144 @@ specified in ``$HOME/.teuthology.yaml``::
+    +----+------------+--------+------------+-------------+-------------------------+
+    +----+------------+--------+------------+-------------+-------------------------+
+* create a passwordless ssh public key with::
+
+    $ openstack keypair create myself > myself.pem
+    +-------------+-------------------------------------------------+
+    | Field       | Value                                           |
+    +-------------+-------------------------------------------------+
+    | fingerprint | e0:a3:ab:5f:01:54:5c:1d:19:40:d9:62:b4:b3:a1:0b |
+    | name        | myself                                          |
+    | user_id     | 5cf9fa21b2e9406b9c4108c42aec6262                |
+    +-------------+-------------------------------------------------+
+    $ chmod 600 myself.pem
+
+Usage
+-----
+
+* Create a passwordless ssh public key::
+
+    $ openstack keypair create myself > myself.pem
+    $ chmod 600 myself.pem
+
+* Run the dummy suite (it does nothing useful but shows all works as
+  expected)::
+
+    $ teuthology-openstack --key-filename myself.pem --key-name myself --suite dummy
+    Job scheduled with name ubuntu-2015-07-24_09:03:29-dummy-master---basic-openstack and ID 1
+    2015-07-24 09:03:30,520.520 INFO:teuthology.suite:ceph sha1: dedda6245ce8db8828fdf2d1a2bfe6163f1216a1
+    2015-07-24 09:03:31,620.620 INFO:teuthology.suite:ceph version: v9.0.2-829.gdedda62
+    2015-07-24 09:03:31,620.620 INFO:teuthology.suite:teuthology branch: master
+    2015-07-24 09:03:32,196.196 INFO:teuthology.suite:ceph-qa-suite branch: master
+    2015-07-24 09:03:32,197.197 INFO:teuthology.repo_utils:Fetching from upstream into /home/ubuntu/src/ceph-qa-suite_master
+    2015-07-24 09:03:33,096.096 INFO:teuthology.repo_utils:Resetting repo at /home/ubuntu/src/ceph-qa-suite_master to branch master
+    2015-07-24 09:03:33,157.157 INFO:teuthology.suite:Suite dummy in /home/ubuntu/src/ceph-qa-suite_master/suites/dummy generated 1 jobs (not yet filtered)
+    2015-07-24 09:03:33,158.158 INFO:teuthology.suite:Scheduling dummy/{all/nop.yaml}
+    2015-07-24 09:03:34,045.045 INFO:teuthology.suite:Suite dummy in /home/ubuntu/src/ceph-qa-suite_master/suites/dummy scheduled 1 jobs.
+    2015-07-24 09:03:34,046.046 INFO:teuthology.suite:Suite dummy in /home/ubuntu/src/ceph-qa-suite_master/suites/dummy -- 0 jobs were filtered out.
+
+    2015-07-24 11:03:34,104.104 INFO:teuthology.openstack:
+    web interface: http://167.114.242.13:8081/
+    ssh access   : ssh ubuntu@167.114.242.13 # logs in /usr/share/nginx/html
+
+* Visit the web interface (the URL is displayed at the end of the
+  teuthology-openstack output) to monitor the progress of the suite.
+
+* The virtual machine running the suite will persist for forensic
+  analysis purposes. To destroy it run::
+
+    $ teuthology-openstack --key-filename myself.pem --key-name myself --teardown
+
+* The test results can be uploaded to a publicly accessible location
+  with the ``--upload`` flag::
```

Comment:
Apparently this is a flag you want to pass when scheduling; that's confusing since it's mentioned *after* the docs explaining how to create and destroy a cluster. This should also mention that ``--upload`` by itself won't be useful unless ``--archive-upload`` has a useful value.